File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> count;
        for (int num : nums) {
            count[num]++;
        }

        vector<pair<int, int>> freqVec(count.begin(), count.end());
        sort(freqVec.begin(), freqVec.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
            return a.second > b.second; // sort by frequency descending
        });

        vector<int> result;
        for (int i = 0; i < k && i < freqVec.size(); ++i) {
            result.push_back(freqVec[i].first);
        }
        return result;
    }
};

File: Data Structures & Algorithms/find-anagram-mappings/submission-0.py
class Solution:
    def anagramMappings(self, nums1: List[int], nums2: List[int]) -> List[int]:
        m = {}

        for i in range(len(nums1)):
            m[nums2[i]]=i
        
        result=[]
        for num in nums1:
            result.append(m[num])


        return result